zkvm/src/vm.rs

```diff
@@ -579,6 +583,35 @@ where
         })
     }
 
+    fn delegate(&mut self) -> Result<(), VMError> {
+        // Signature
+        let sig = self.pop_item()?.to_data()?.to_bytes();
+        if sig.len() != 64 {
+            return Err(VMError::FormatError);
+        }
+        let signature = Signature::from_bytes(SliceReader::parse(&sig, |r| r.read_u8x64())?)?;
+
+        // Program
+        let prog = self.pop_item()?.to_data()?;
+        let prog_bytes = prog.clone().to_bytes();
+
+        // Verification key
+        let contract = self.pop_item()?.to_contract()?;
```

Comment:
OOPS, my bad - there's a bug in the spec - we need to place all payload of the contract onto the stack before we continue execution of the new program. Could you please add this step to the spec and change the code accordingly?

```
        for item in contract.payload.into_iter() {
            self.push_item(item);
        }
```

Reply:
Yep sure!